tokio-console/src/main.rs

```diff
@@ -214,3 +223,24 @@ async fn watch_details_stream(
         }
     }
 }
+
+fn gen_completion(install: bool, shell: Shell) -> color_eyre::Result<()> {
+    let mut app = config::Config::command();
+    let mut buf: Box<dyn std::io::Write> = if install {
+        let mut home_dir = dirs::home_dir()
+            .ok_or_else(|| color_eyre::eyre::eyre!("fail to find home directory"))?;
+        match shell {
+            Shell::Zsh => {
+                home_dir.push(".zsh_functions/_tokio_console");
+                let f = std::fs::File::create(&home_dir)
+                    .wrap_err_with(|| format!("fail to open {}", home_dir.display()))?;
+                Box::new(std::io::BufWriter::new(f))
+            }
+            _ => color_eyre::eyre::bail!("Not support to install completion script on {}", shell),
+        }
+    } else {
+        Box::new(std::io::stdout())
+    };
+    generate(shell, &mut app, "tokio-console", &mut buf);
```

Comment:
nit, take it or leave it: i would probably not import the `generate` function into this file globally, as it's not _super_ clear what's being generated this way. i would probably have written
```suggestion
    clap_complete::generate(shell, &mut app, "tokio-console", &mut buf);
```

but, it's not a big deal either way.

Reply:
fixed: https://github.com/tokio-rs/console/pull/336/commits/0c02ebfce52d1989034aa4ac140df64f12999bec